### PR TITLE
Fix broken export comment

### LINKED ISSUE #885: Invalid code generated for weird export with comment
```js
export //comment
{}
```

[turns into](https://prettier.github.io/prettier/#%7B%22content%22%3A%22export%20%2F%2Fcomment%5Cn%7B%7D%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A801%2C%22tabWidth%22%3A1%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22doc%22%3Afalse%7D%7D):

```js
export //comment{};
```

[which gives this error](https://prettier.github.io/prettier/#%7B%22content%22%3A%22export%20%2F%2Fcomment%7B%7D%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A801%2C%22tabWidth%22%3A1%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22doc%22%3Afalse%7D%7D):

```
SyntaxError: Unexpected token, expected { (2:0)
  1 | export //comment{};
> 2 | 
    | ^
```

### PR DESCRIPTION
**Pull request by yamafaktory**
This should fix #885.

### CODE REVIEW

File: src/comments.js [hunk 1]
```diff
@@ -575,6 +576,16 @@ function handlePropertyComments(enclosingNode, comment) {
   return false;
 }
 
+function handleExportNamedDeclarationComments(enclosingNode, comment) {
+  if (enclosingNode && enclosingNode.type === "ExportNamedDeclaration") {
+    // Enforce all comments to be leading block comments.
+    comment.type = "CommentBlock";
```
Comment (yamafaktory) [inline src/comments.js]: We are already doing it for template literals.

Comment (vjeux) [inline src/comments.js]: This is not the right fix here, we should just make the comment leading

Comment (yamafaktory) [inline src/comments.js]: Ok, I was not sure about that one and I found it weird at first glance when I tried it.

**Comment by yamafaktory** [comment]
@vjeux fixed!

**Comment by vjeux** [comment]
For context, all of the reports from @lydell have been found via fuzzing, so they are unlikely to come up in real life. We don't want to spend a lot of time making them pretty but we want to make sure that they are stable.

**Pull request merged**